src/slic3r/GUI/GLCanvas3D.cpp

```diff
@@ -3690,15 +3690,18 @@ void GLCanvas3D::on_mouse_wheel(wxMouseEvent& evt)
     double direction_factor = 1.0;
 #endif
     auto delta = direction_factor * (double)evt.GetWheelRotation() / (double)evt.GetWheelDelta();
+    auto cnv_size = get_canvas_size();
+    auto screen_center_3d_pos = _mouse_to_3d({ cnv_size.get_width() * 0.5, cnv_size.get_height() * 0.5 });
+    auto mouse_3d_pos = _mouse_to_3d({evt.GetX(), evt.GetY()});
+    auto displacement = mouse_3d_pos - screen_center_3d_pos;
     bool zoom_to_mouse = wxGetApp().app_config->get("zoom_to_mouse") == "true";
-    if (!zoom_to_mouse) {// zoom to center
+
+    if (!zoom_to_mouse || !displacement.allFinite() || displacement.maxCoeff() > 1000 || displacement.minCoeff() < -1000) {
```

Comment:
I really don't like compare value twice, but I couldn't find better way to check absolute value limits.